shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/dml/impl/ShardingInsertStatementValidator.java

```diff
@@ -84,6 +103,94 @@ private boolean isContainsKeyGenerateColumn(final ShardingRule shardingRule, fin
+            && isSameValue(shardingConditionValue1, shardingConditionValue2);
+    }
+
+    private boolean isSameLogicTable(final ShardingRule shardingRule, final ShardingConditionValue shardingValue1, final ShardingConditionValue shardingValue2) {
+        return shardingValue1.getTableName().equals(shardingValue2.getTableName()) || isBindingTable(shardingRule, shardingValue1, shardingValue2);
+    }
+
+    private boolean isBindingTable(final ShardingRule shardingRule, final ShardingConditionValue shardingValue1, final ShardingConditionValue shardingValue2) {
+        Optional<BindingTableRule> bindingRule = shardingRule.findBindingTableRule(shardingValue1.getTableName());
+        return bindingRule.isPresent() && bindingRule.get().hasLogicTable(shardingValue2.getTableName());
+    }
+
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    private boolean isSameValue(final ShardingConditionValue shardingConditionValue1, final ShardingConditionValue shardingConditionValue2) {
+        if (shardingConditionValue1 instanceof ListShardingConditionValue && shardingConditionValue2 instanceof ListShardingConditionValue) {
+            return SafeNumberOperationUtils.safeCollectionEquals(
+                ((ListShardingConditionValue) shardingConditionValue1).getValues(), ((ListShardingConditionValue) shardingConditionValue2).getValues());
+        } else if (shardingConditionValue1 instanceof RangeShardingConditionValue && shardingConditionValue2 instanceof RangeShardingConditionValue) {
+            return SafeNumberOperationUtils.safeRangeEquals(
+                ((RangeShardingConditionValue) shardingConditionValue1).getValueRange(), ((RangeShardingConditionValue) shardingConditionValue2).getValueRange());
+        }
+        return false;
+    }
+
+    private ShardingConditions createShardingConditions(final SQLStatementContext<InsertStatement> sqlStatementContext,
+        final List<Object> parameters, final ShardingSphereSchema schema, final ShardingRule rule) {
+        List<ShardingCondition> shardingConditions;
+        if (sqlStatementContext.getSqlStatement() instanceof DMLStatement) {
+            ShardingConditionEngine shardingConditionEngine = new WhereClauseShardingConditionEngine(rule, schema);
+            shardingConditions = shardingConditionEngine.createShardingConditions(sqlStatementContext, parameters);
+        } else {
+            shardingConditions = Collections.emptyList();
+        }
+        return new ShardingConditions(shardingConditions);
+    }
+
+    private boolean isNeedMergeShardingValues(final SQLStatementContext<?> sqlStatementContext, final ShardingRule rule) {
+        boolean insertSelectContainsSubquery = sqlStatementContext instanceof InsertStatementContext && null != ((InsertStatementContext) sqlStatementContext).getInsertSelectContext()
+            && ((InsertStatementContext) sqlStatementContext).getInsertSelectContext().getSelectStatementContext().isContainsSubquery();
```

Comment:
I think this check is redundant because we've already judged `insertSelectSegment.isPresent()`, and we can use `insertSelectStatement` directly to determine whether sub-queries are included. BTW, this method can also be extracted into the `ShardingDMLStatementValidator` class. What do you think ?

Reply:
I think to move this method to the ShardingDMLStatementValidator is better.